File: game/src/test/java/application/browser/PlayerActiveManaHelper.java
package application.browser;

import com.matag.cards.properties.Color;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;
import java.util.stream.Collectors;

public class PlayerActiveManaHelper {

  private static final Logger LOGGER = LoggerFactory.getLogger(PlayerActiveManaHelper.class);

  private MatagBrowser matagBrowser;

  PlayerActiveManaHelper(MatagBrowser matagBrowser) {
    this.matagBrowser = matagBrowser;
  }

  public void toHaveMana(List<Color> colors) {
    matagBrowser.wait(driver -> {
      List<WebElement> imgs = getElementContainer().findElements(By.tagName("img"));
      List<String> alts = imgs.stream().map((webElement -> webElement.getAttribute("alt"))).sorted().collect(Collectors.toList());
      List<String> expectedColors = colors.stream().map(Enum::name).sorted().collect(Collectors.toList());
      LOGGER.info("alts={}   expectedColors={}", alts, expectedColors);
      return expectedColors.equals(alts);
    });
  }

  private WebElement getElementContainer() {
    return matagBrowser.findElement(By.id("player-active-mana"));
  }
}

File: src/test/kotlin/application/browser/PlayerActiveManaHelper.kt
package application.browser

import com.matag.cards.properties.Color
import org.openqa.selenium.By
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.ui.ExpectedCondition
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class PlayerActiveManaHelper internal constructor(private val matagBrowser: MatagBrowser) {
    fun toHaveMana(colors: List<Color>) {
        matagBrowser.wait(ExpectedCondition {
            val imgs = elementContainer().findElements(By.tagName("img"))
            val alts = imgs.map { it.getAttribute("alt")!! }.sorted()
            val expectedColors = colors.map { it.name }.sorted()
            LOGGER.info("alts={}   expectedColors={}", alts, expectedColors)
            expectedColors == alts
        })
    }

    private fun elementContainer(): WebElement =
        matagBrowser.findElement(By.id("player-active-mana"))

    companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(PlayerActiveManaHelper::class.java)
    }
}
